crates/cli/src/config.rs

```diff
@@ -33,10 +59,29 @@ impl ServerConfig {
     }
 }
 
-#[derive(Default, Deserialize, Serialize, Debug, Clone)]
+impl From<&Table> for ServerConfig {
+    fn from(table: &Table) -> Self {
+        let nickname = table.get(NICKNAME_KEY).and_then(Item::as_str).map(String::from);
+        let host = table.get(HOST_KEY).and_then(Item::as_str).map(String::from).unwrap();
+        let protocol = table
+            .get(PROTOCOL_KEY)
+            .and_then(Item::as_str)
+            .map(String::from)
+            .unwrap();
+        let ecdsa_public_key = table.get(ECDSA_PUBLIC_KEY).and_then(Item::as_str).map(String::from);
+        ServerConfig {
+            nickname,
+            host,
+            protocol,
+            ecdsa_public_key,
+        }
+    }
+}
```

Comment:
Should `ServerConfig` really be `From<&Table>`? Can we convert *any* `Table` into a meaningful `ServerConfig`? It seems to me that this should more likely be a `TryFrom` impl, though I am not sure. E.g. what happens if I do:
```rust
let table = toml::Table::new();
let server_config = ServerConfig::from(&table);
```